SYCL/Basic/group_async_copy.cpp

```diff
@@ -122,22 +122,22 @@ template <typename T> int test(size_t Stride) {
        size_t Offset = GrId * WorkGroupSize;
        if (Stride == 1) { // Check the version without stride arg.
          auto E = NDId.async_work_group_copy(
-             Local.get_pointer(), In.get_pointer() + Offset, NElemsToCopy);
+             local_ptr<T>(Local), In.get_pointer() + Offset, NElemsToCopy);
```

Comment:
IIUC, this uses 

```
template <typename ElementType,
          access::decorated IsDecorated = access::decorated::legacy>
using local_ptr
```

deprecated value (`legacy`) for `access::decorated` template param. Is that possible to use either `yes` or `no` instead?

Reply:
@aelovikov-intel, Sounds reasonable, I would suggest doing so in another PR, though.

Reply:
Well, one can argue that such a change would be independent of your intel/llvm PR and consequently make this one redundant as well.

Also, the spec for `async_work_group_copy` says it only accepts ` decorated_local_ptr` so I'm not even sure what exactly happens here. Do we have an implicit conversion happening?

I guess I can give you an LGTM, but still...